### PR TITLE
Add Notebook Docs

### PR DESCRIPTION
**Pull request by nstogner**
* Rename Architecture to Overview (it was not really an architecture page)
* Add Notebook page
* Add Basic Notebook Guide
* Support conditional output rendering based on tags (we can now leave all ran-cells in place and declare which ones we want to include in the website)
* Update output formatting (attempting to make it clearer that it is execution output)
* Small tweak to a diagram

### CODE REVIEW

File: docs/notebooks-b.md [hunk 1]
```diff
@@ -0,0 +1,42 @@
+# Notebooks
+
+Jupyter notebooks sit at the core of machine learning expeditions at Substratus, the driver seat on your journey of refining raw data into a finely-tuned, deployed model.
```
Comment (samos123) [inline docs/notebooks-b.md:3]: this seems incorrect. I think we should go with notebook-a

File: docs/notebooks-b.md [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+## Key Features
+
+**1. Emulated Environments.**
+
+Imagine meeting your doppelganger, a perfect replica of yourself who thinks, acts, and even makes the same errors just as you would. Those are exactly the traits Substratus notebooks possess. They replicate the ML environments where your tasks come alive using the consistency provided by containers.
```
Comment (samos123) [inline docs/notebooks-b.md:9]: seems to use too many words to get the poitn across

File: docs/notebooks-b.md (hunk 1 above)
Comment (brandonjbjelland) [reply docs/notebooks-b.md:3]: ML expeditions? I'm looking for less of a journey and more of a jaunt, personally.

File: docs/notebooks-b.md (hunk 2 above)
Comment (brandonjbjelland) [reply docs/notebooks-b.md:9]: +1 and I don't really think the analogy works either

File: docs/notebooks-b.md [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+Whether it's model-training code you're refining or a dataset you're importing, your development space matches your final execution platform - identical Python version, Python packages, and the version of tools. It's like perfecting your symphony in the same auditorium where you're set to perform.
+
+**2. Data Integrated.**
+
+Often, datasets and models are massive behemoths. Substratus notebooks run remotely, with these titans mounted as read-only filesystems. The outcome? You can scrutinize and toy with bulky models and datasets without the patience-testing process of downloading them to your local machine.
```
Comment (brandonjbjelland) [inline docs/notebooks-b.md:15]: "massive behemoths" - cmon ChatGPT

File: docs/notebooks-b.md [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+
+**3. Remote-Local Syncing.**
+
+With just one command, teleport your local directory to the cluster. Substratus sets to work, building your containerized notebook environment. You are presented with a notebook in your browser, all of your files and packages ready to go, with any changes you make showing up almost immediately back on your machine.
```
Comment (brandonjbjelland) [inline docs/notebooks-b.md:19]: I dont think this speaks well to an engineering audience.

File: docs/notebooks-b.md [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+
+## Typical Workflow
+
+A typical development workflow on Substratus that leverages notebooks might resemble a tango of intuitive steps:
```
Comment (brandonjbjelland) [inline docs/notebooks-b.md:23]: "a tango of intuitive steps"? Paul Graham should be in protest.

**Comment by samos123** [comment]
Sounds like Brandon and me both think option a is the way to go

**Comment by brandonjbjelland** [comment]
C is a polished version of A but often wordy in a way that doesn't add value. I would take some of the polish but mostly stick with A.

B is ridiculous the closer you read it.

**Comment by nstogner** [comment]
Issue added to track missing tests: #38

File: docs/walkthrough/intro-to-notebooks.md [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+With the following command Substratus will upload, build, and serve your notebook. The `-d` flag instructs the command to launch from the current directory. Once launched, try making a change to a file in the notebook and saving it.
+
+
+```bash
+kubectl notebook -d .
+```
+
+
+
+Substratus continuously syncs any changes you made in your remote notebook back to your local directory.
```
Comment (samos123) [inline docs/walkthrough/intro-to-notebooks.md:60]: I thought you have to use `--sync` or it to sync?

**Review by samos123 — APPROVED** [review]

File: docs/notebooks.md [hunk 7]
```diff
@@ -0,0 +1,34 @@
+# Notebooks
```
Comment (samos123) [inline docs/notebooks.md:1]: Should you add something like this at the top?
```
---
sidebar_position: 6
---
```

File: Makefile [hunk 8]
```diff
@@ -33,3 +33,23 @@ freeze:
...
+.PHONY: clear-notebooks
+clear-notebooks:
+	./nbconvert/clear-all.sh
+
+.PHONY: convert-notebooks
+convert-notebooks:
+	./nbconvert/convert-all.sh
+
+.PHONY: prepare-release
+prepare-release: convert-notebooks
```
Comment (brandonjbjelland) [inline Makefile:54]: We probably want both steps here, right?
```suggestion
prepare-release: clear-notebooks convert-notebooks
```

Comment (nstogner) [reply Makefile:54]: Not needed here, I changed nbconvert to only include outputs that the notebook cells declare they would like to output with tags.

File: docs/walkthrough/intro-to-notebooks.md (hunk 6 above)
Comment (nstogner) [reply docs/walkthrough/intro-to-notebooks.md:60]: `-d` turns on `--sync` by default

**Pull request merged**